src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -63,37 +65,509 @@ internal partial class InvertIfCodeRefactoringProvider : CodeRefactoringProvider
+                                return SyntaxFactory.ReturnStatement();
+                            case var syntaxKind:
+                                throw ExceptionUtilities.UnexpectedValue(syntaxKind);
+                        }
+                    }
+
+                case InvertIfStyle.WithSubsequenceExitPoint:
+                    {
+                        var newIfBody = (StatementSyntax)subsequenceExitPoint;
+                        var updatedIf = ifNode.WithCondition(negatedCondition)
+                            .WithStatement(SyntaxFactory.Block(newIfBody));
+
+                        var currentParent = ifNode.Parent;
+                        var statements = GetStatements(currentParent);
+                        var index = statements.IndexOf(ifNode);
+
+                        var statementsBeforeIf = statements.Take(index);
+
+                        var updatedParent = WithStatements(currentParent, statementsBeforeIf.Concat(updatedIf).Concat(UnwrapBlock(ifNode.Statement)).Concat(newIfBody))
+                            .WithAdditionalAnnotations(Formatter.Annotation);
+
+                        return document.WithSyntaxRoot(root.ReplaceNode(currentParent, updatedParent));
+                    }
+
+                case InvertIfStyle.MoveSubsequenceToElseBody:
+                    {
+                        var currentParent = ifNode.Parent;
+                        var statements = GetStatements(currentParent);
+                        var index = statements.IndexOf(ifNode);
+
+                        var statementsBeforeIf = statements.Take(index);
+                        var statementsAfterIf = statements.Skip(index + 1).ToArray();
+
+                        var updatedIf = ifNode
+                            .WithCondition(negatedCondition)
+                            .WithStatement(ReplaceEmbeddedStatement(ifNode.Statement, statementsAfterIf));
+
+                        var updatedParent = WithStatements(currentParent, statementsBeforeIf.Concat(updatedIf))
+                            .WithAdditionalAnnotations(Formatter.Annotation);
+
+                        return document.WithSyntaxRoot(root.ReplaceNode(currentParent, updatedParent));
+                    }
+
+                case InvertIfStyle.WithElseClause:
+                    {
+                        var currentParent = ifNode.Parent;
+                        var statements = GetStatements(currentParent);
+                        var index = statements.IndexOf(ifNode);
+
+                        var statementsBeforeIf = statements.Take(index);
+                        var statementsAfterIf = statements.Skip(index + 1).ToArray();
+
+                        var updatedIf = ifNode
+                            .WithCondition(negatedCondition)
+                            .WithStatement(ReplaceEmbeddedStatement(ifNode.Statement, statementsAfterIf))
+                            .WithElse(SyntaxFactory.ElseClause(ifNode.Statement));
```

Comment:
I don't think this will work for this case:

    void M()
    {
        if (c)
        {
            f();
        }
        g();
        g();
    }

Reply:
@chborl Test added, see #26793. I'll start VB part once C# is reviewed.